main/src/ca/uwaterloo/flix/language/phase/Parser.scala

```diff
@@ -631,6 +631,16 @@ class Parser(val source: Source) extends org.parboiled2.Parser {
       }
     }
 
+    def Select: Rule1[ParsedAst.Expression.Select] = {
+      def Rule: Rule1[ParsedAst.SelectRule] = rule {
+        atomic("case") ~ WS ~ Names.Variable ~ optWS ~ "=" ~ optWS ~ "<-" ~ optWS ~ Expression ~ optWS ~ atomic("=>") ~ optWS ~ Expression ~> ParsedAst.SelectRule
```

Comment:
Maybe we should just simplify it to: `case x <- chan => 42`. I think that looks neat enough.